File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_word(string text);
int count_sentence(string text);


int main(void){
    string name = get_string("Text: ");
    double letra = count_letters(name);
    double word = count_word(name);
    double sentence = count_sentence(name);
    /*printf("O numero de letas é: %d\n",letra);
    printf("O numero de word é: %d\n",word);
    printf("O numero de sentence é: %d\n",sentence);*/
    double L = (letra*100)/word;
    double S = (sentence*100)/word;
    double index = 0.0588 * L - 0.296 * S - 15.8;
    // printf("%.2f",index);
    int resp = (int)index;
    double veri = index-resp;
    double final = 0;
    if(veri>0.5)
         final = ceil(index);
    }
    else if(veri<=0.5){
         final = floor(index);
     }

     resp = final;

    // printf("%d", resp);

    if(resp< 1){
        printf("Before Grade 1\n");
    }
    else if(resp >16){
        printf("Grade 16+\n");
    }
    else{
        printf("Grade %d\n", resp);
    }


}

int count_letters(string text){
    int letra=0;
    for(int cont = 0; cont<strlen(text);cont++){
        if(isalpha(text[cont])){
            letra++;
        }
        else{
            continue;
        }
    }
    return letra;
}

int count_word(string text){
    int word=1;;
    char c = ' ';
    for(int cont = 0; cont<strlen(text);cont++){
        if(text[cont]==c){
            word++;
        }
        else{
            continue;
        }
    }
    return word;
}

int count_sentence(string text){
    int sentence=0;;
    char c = '.';
    for(int cont = 0; cont<strlen(text);cont++){
        if(text[cont]==c||text[cont]=='?'||text[cont]=='!'){
            sentence++;
        }
        else{
            continue;
        }
    }
    return sentence;
}

File: sentimental-readability/readability.py
from cs50 import get_string

def count_letters(text):
    letra = 0
    for contador in text:
        if(contador.isalpha()):
            letra += 1
        else:
            continue
    return letra

def count_word(text):
    word=1
    c = ' '
    for cont in range(0,len(text)):
        if(text[cont]==c):
            word += 1
        else:
            continue
    return word

def count_sentence(text):
    sentence = 0
    for cont in range(0,len(text)):
        if((text[cont]=='.') or (text[cont]=='?') or (text[cont]=='!')):
            sentence += 1
        else:
            continue
    return sentence

text = get_string("Text: ")
letra = count_letters(text)
word = count_word(text)
sentence = count_sentence(text)
#print("O número de letras = %d" % letra)
#print("O número de palavras = %d" % word)
#print("O número de sentenças = %d" % sentence)
L = (letra*100)/word
S = (sentence*100)/word
index = 0.0588 * L - 0.296 * S - 15.8
if(index<1):
    print("Before Grade: 1")
elif(index>16):
    print("Grade: 16+")
else:
    print("Grade: %d"%round(index))